### PR TITLE
refactor(api)!: use singular/plural consistently in the autocmd API

### PR DESCRIPTION
**Pull request by ggandor**
Follow-up to https://github.com/neovim/neovim/pull/17854#issuecomment-1081823407.

### CODE REVIEW

File: runtime/doc/api.txt [hunk 1]
```diff
@@ -3310,8 +3310,9 @@ nvim_del_autocmd({id})                                    *nvim_del_autocmd()*
                 See also: ~
                     |nvim_create_autocmd()|
 
-nvim_exec_autocmd({event}, {*opts})                      *nvim_exec_autocmd()*
-                Execute an autocommand |autocmd-execute|.
+nvim_exec_autocmds({event}, {*opts})                    *nvim_exec_autocmds()*
+                Execute all autocommands for {event}, that match the
```
Comment (clason) [inline runtime/doc/api.txt:3314]:
```suggestion
                Execute all autocommands for {event} that match the
```

File: src/nvim/api/autocmd.c [hunk 2]
```diff
@@ -697,7 +697,8 @@ void nvim_del_augroup_by_name(String name)
   augroup_del(name.data, false);
 }
 
-/// Execute an autocommand |autocmd-execute|.
+/// Execute all autocommands for {event}, that match the corresponding
```
Comment (clason) [inline src/nvim/api/autocmd.c:700]:
```suggestion
/// Execute all autocommands for {event} that match the corresponding
```

File: src/nvim/api/autocmd.c [hunk 3]
```diff
@@ -697,7 +697,8 @@ void nvim_del_augroup_by_name(String name)
   augroup_del(name.data, false);
 }
 
-/// Execute an autocommand |autocmd-execute|.
+/// Execute all autocommands for {event} that match the corresponding
+//  {opts} |autocmd-execute|.
```
Comment (clason) [inline src/nvim/api/autocmd.c:701]:
```suggestion
///  {opts} |autocmd-execute|.
```

Comment (clason) [reply src/nvim/api/autocmd.c:701]: (CI complains about that comment not matching the committed API.)

**Review by justinmk — APPROVED** [review]

**Pull request merged**